cirq/ops/three_qubit_gates.py

```diff
@@ -145,6 +144,125 @@ def __str__(self) -> str:
         return 'CCZ**{}'.format(self._exponent)
 
 
+class ThreeQubitDiagonalGate(gate_features.ThreeQubitGate):
+    """A gate given by a diagonal 8x8 matrix."""
+
+    def __init__(self,
+                 a: float = 0.0,
```

Comment:
It's a bit silly to take this many arguments instead of just a list of floats. Also, make sure to include `radians` in the name, and document that it's in radians.

Reply:
Done